LEGO1/lego/legoomni/src/build/legocarbuildpresenter.cpp

```diff
@@ -185,23 +230,125 @@ void LegoCarBuildAnimPresenter::EndAction()
 // FUNCTION: BETA10 0x1007151e
 void LegoCarBuildAnimPresenter::FUN_10079050(MxS16 p_index)
 {
-	FUN_10079090(m_unk0x128[p_index].m_unk0x04, m_unk0x128[p_index].m_unk0x00);
-	FUN_100795d0(m_unk0x128[p_index].m_unk0x04);
+	SwapNodesByName(m_parts[p_index].m_wiredName, m_parts[p_index].m_name);
+	FUN_100795d0(m_parts[p_index].m_wiredName);
 }
 
-// STUB: LEGO1 0x10079090
-// STUB: BETA10 0x10071584
-void LegoCarBuildAnimPresenter::FUN_10079090(LegoChar* p_param1, LegoChar* p_param2)
+// FUNCTION: LEGO1 0x10079090
+// FUNCTION: BETA10 0x10071584
+void LegoCarBuildAnimPresenter::SwapNodesByName(LegoChar* p_name1, LegoChar* p_name2)
 {
-	// TODO
+	char buffer[40];
+
+	if (stricmp(p_name1, p_name2) != 0) {
+		LegoAnimNodeData* node1 = FindNodeDataByName(m_anim->GetRoot(), p_name1);
+		LegoAnimNodeData* node2 = FindNodeDataByName(m_anim->GetRoot(), p_name2);
+
+		strcpy(buffer, node1->GetName());
+		strcpy(node1->GetName(), node2->GetName());
+		strcpy(node2->GetName(), buffer);
+
+		LegoU16 val1 = node1->GetUnknown0x20();
+		node1->SetUnknown0x20(node2->GetUnknown0x20());
+		node2->SetUnknown0x20(val1);
+	}
 }
 
-// STUB: LEGO1 0x10079160
-// STUB: BETA10 0x1007165d
+// FUNCTION: LEGO1 0x10079160
+// FUNCTION: BETA10 0x1007165d
 void LegoCarBuildAnimPresenter::FUN_10079160()
 {
-	// called from LegoCarBuildAnimPresenter::StreamingTickle()
-	// TODO
+	LegoTreeNode* root;
+	LegoAnimNodeData* data2;
+	MxS16 i;
+	MxS16 totalNodes = CountTotalTreeNodes(m_anim->GetRoot());
+	LegoChar* name;
+	LegoTreeNode* destNode;
+	LegoAnimNodeData* destData;
+	LegoTreeNode** children;
```

Comment:
I couldn't get this stack layout quite right. What I observed:
* The order of declaration does not appear to matter when the variables are not initialized
* The names of these variables **do** matter, at least in BETA10. However, the pattern is still not quite clear to me
* The place where the variable is declared does not appear to matter as long as it is on the top level (it does matter whether the variable is declared in a smaller scope)
* Tests with my preliminary tooling suggest that these variables map 1:1 to the variables in BETA10, suggesting that the logic is correct.